toonz/sources/toonz/shortcutpopup.cpp

```diff
@@ -415,4 +471,320 @@ void ShortcutPopup::onSearchTextChanged(const QString &text) {
+  }
+  preset.endGroup();
+  m_sViewer->shortcutChanged();
+  m_dialog->hide();
+  buildPresets();
+  setCurrentPresetPref(QString::fromStdString(fp.getName()));
+}
+
+//-----------------------------------------------------------------------------
+
+bool ShortcutPopup::showConfirmDialog() {
+  QString question(tr("This will overwrite all current shortcuts. Continue?"));
+  int ret =
+      DVGui::MsgBox(question, QObject::tr("OK"), QObject::tr("Cancel"), 0);
+  if (ret == 0 || ret == 2) {
+    // cancel (or closed message box window)
+    return false;
+  } else
+    return true;
+}
+
+//-----------------------------------------------------------------------------
+
+bool ShortcutPopup::showOverwriteDialog(QString name) {
+  QString question(tr("A file named ") + name +
+                   tr(" already exists.  Do you want to replace it?"));
+  int ret = DVGui::MsgBox(question, QObject::tr("Yes"), QObject::tr("No"), 0);
+  if (ret == 0 || ret == 2) {
+    // cancel (or closed message box window)
+    return false;
+  } else
+    return true;
+}
+
+//-----------------------------------------------------------------------------
+
+void ShortcutPopup::showDialog(QString text) {
+  if (m_dialog == NULL) {
+    m_dialogLabel = new QLabel("", this);
+    m_dialog      = new DVGui::Dialog(this, false, false);
+    m_dialog->setWindowTitle(tr("OpenToonz - Setting Shortcuts"));
+    m_dialog->setModal(false);
+
+    m_dialog->setTopMargin(10);
+    m_dialog->setTopSpacing(10);
+    m_dialog->setLabelWidth(500);
+    m_dialog->beginVLayout();
+    m_dialog->addWidget(m_dialogLabel, false);
+    m_dialog->endVLayout();
+  }
+  m_dialogLabel->setText(text);
+  m_dialog->show();
+}
+
+//-----------------------------------------------------------------------------
+
+void ShortcutPopup::onExportButton(TFilePath fp) {
+  if (fp == TFilePath()) {
+    m_saveShortcutsPopup = new GenericSaveFilePopup("Save Current Shortcuts");
+    m_saveShortcutsPopup->addFilterType("ini");
+    fp = m_saveShortcutsPopup->getPath();
+    if (fp == TFilePath()) return;
+  }
+  showDialog("Saving Shortcuts");
+  QString shortcutString = "[shortcuts]\n";
+  for (int commandType = UndefinedCommandType; commandType <= MenuCommandType;
+       commandType++) {
+    std::vector<QAction *> actions;
+    CommandManager::instance()->getActions((CommandType)commandType, actions);
+    for (QAction *action : actions) {
+      qApp->processEvents();
+      std::string id = CommandManager::instance()->getIdFromAction(action);
+      std::string shortcut =
+          CommandManager::instance()->getShortcutFromAction(action);
+      shortcutString = shortcutString + QString::fromStdString(id) + "=" +
+                       QString::fromStdString(shortcut) + "\n";
+    }
+  }
+  QFile file(fp.getQString());
+  file.open(QIODevice::WriteOnly | QIODevice::Text);
+  QTextStream out(&file);
+  out << shortcutString;
+  file.close();
```

Comment:
I think using `QSettings` would be easier to maintain.
Please refer to the function `CommandManager::setShortcut()` at the line 97 of `menubarcommand.cpp`.

Furthermore, how about to skip the commands with no shortcut and save the commands with shortcuts only?